Add field name mapping for options.attributes in Instance.increment

lostintime opened the pull request:
Got same issue as https://github.com/sequelize/sequelize/issues/4389, with model like: 

```
const Article = sequelize.define(
    'articles',
    {
      id: {
        field: 'id',
        type: DataTypes.BIGINT,
        primaryKey: true,
        autoIncrement: true,
      },
      createdAt: {
        field: 'created_at',
        type: DataTypes.DATE,
        index: true,
        allowNull: false,
        defaultValue: DataTypes.NOW,
      },
      commentsCount: {
        field: 'comments_count',
        type: DataTypes.BIGINT,
        allowNull: false,
        defaultValue: 0,
      },
      updatedAt: {
        field: 'updated_at',
        type: DataTypes.DATE,
        allowNull: true,
      },
    }, {
      underscored: true,
      createdAt: 'createdAt',
      updatedAt: 'updatedAt',
    });
```

while trying to increment a field:

```
Article.increment('commentsCount')
```

Looks like fields mapping is missing for options.attributes in Instance.increment method.

janmeier commented:
[`mapOptionFieldNames`](https://github.com/sequelize/sequelize/blob/f274ebaf4eeb1ef80ae27ded91e6b2c65bc97a2f/lib/utils.js#L127).

Also, please include an integration test :)

janmeier commented:
Ping @lostintime

lostintime commented:
@janmeier sorry, was a bit busy, let me do this today. Thx.

lostintime commented:
Rebased with master, changed to ES6 style, added 1 integration and 1 unit test case for changed method. @janmeier please don't hesitate to ping me if something is missing or wrong :) thx.

lostintime commented:
@janmeier also - didn't get it about pointed Utils.mapOptionFieldNames method - should I use it? but it supports arrays only for "attributes" property, and looks like is designed for queries, not updates, as in this case.

sushantdhiman commented:
Fixed in https://github.com/sequelize/sequelize/pull/6810
